src/adapt/schema.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as bigquery from '@google-cloud/bigquery';
+import * as protos from '../../protos/protos';
+type StorageTableSchema = protos.google.cloud.bigquery.storage.v1.ITableSchema;
+type StorageTableField =
+  protos.google.cloud.bigquery.storage.v1.ITableFieldSchema;
+type StorageTableFieldType =
+  protos.google.cloud.bigquery.storage.v1.TableFieldSchema.Type;
+
+const StorageTableSchema = protos.google.cloud.bigquery.storage.v1.TableSchema;
+const StorageTableField =
+  protos.google.cloud.bigquery.storage.v1.TableFieldSchema;
+const StorageTableFieldType =
+  protos.google.cloud.bigquery.storage.v1.TableFieldSchema.Type;
+
+const fieldTypeMap: Record<string, StorageTableFieldType> = {
```

Comment:
There are some similar maps at the top of proto.ts, maybe a good split mechanism would be pulling all of the type mapping out into a separate .ts file.